### PR TITLE
Align instanceof check with type cast

### PR DESCRIPTION
**Pull request by sabi0**

### CODE REVIEW

File: lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/standard/processors/AnalyzerQueryNodeProcessor.java [hunk 1]
```diff
@@ -107,10 +105,9 @@ public QueryNode process(QueryNode queryTree) throws QueryNodeException {
   @Override
   protected QueryNode postProcessNode(QueryNode node) throws QueryNodeException {
 
-    if (node instanceof TextableQueryNode
+    if (node instanceof FieldQueryNode
```
Comment (sabi0) [inline lucene/queryparser/src/java/org/apache/lucene/queryparser/flexible/standard/processors/AnalyzerQueryNodeProcessor.java:108]: 'node' is cast below to `FieldQueryNode`. But the condition here was checking for a base type `TextableQueryNode`, excluding some sub-types. If a new `TextableQueryNode` sub-type was added the code below could run into a CCE.

Having the `instanceof FieldQueryNode` condition here makes the type cast below safe. And also eliminates the need to exclude the `RegexpQueryNode` as it does not subclass `FieldQueryNode`.

File: lucene/core/src/java/org/apache/lucene/analysis/tokenattributes/PayloadAttributeImpl.java [hunk 2]
```diff
@@ -62,8 +62,7 @@ public boolean equals(Object other) {
       return true;
     }
 
-    if (other instanceof PayloadAttribute) {
-      PayloadAttributeImpl o = (PayloadAttributeImpl) other;
+    if (other instanceof PayloadAttributeImpl o) {
```
Comment (uschindler) [inline lucene/core/src/java/org/apache/lucene/analysis/tokenattributes/PayloadAttributeImpl.java:65]: This looks like an unrelated change. There is a PR about this type of code change already. It can only be applied to main branch as it requires Java 17.

Comment (uschindler) [reply lucene/core/src/java/org/apache/lucene/analysis/tokenattributes/PayloadAttributeImpl.java:65]: See this PR: #12295

Comment (sabi0) [reply lucene/core/src/java/org/apache/lucene/analysis/tokenattributes/PayloadAttributeImpl.java]: Thank you for linking to the pattern-matching PR.
I've removed the pattern variable part of this change.

**Comment by uschindler** [comment]
I added the changes entry to the existing one for you.

**Comment by sabi0** [comment]
Thank you. I will make sure the new PRs have them out of the box.
